feat(NotificationDrawer): Notification, NotificationDrawer

gilad215 opened the pull request:
<!--
Thanks for your interest in patternfly-react. We appreciate all issues filed and PRs submitted!

Please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What issue is being addressed here?) -->
**What**:
fixes #84
Adding a Notification Drawer.
So far, created all the needed components for rendering the drawer, and a Storybook with an expand knob

<!-- Please provide a link to your fork's Storybook. See README notes on how to do this. -->
**Link to Storybook**:
[Storybook](https://rawgit.com/gilad215/patternfly-react/storybook/feature/notif-drawer/index.html?knob-Expanded%20Drawer=false&selectedKind=NotificationDrawer&selectedStory=Stateful%20Drawer%20Wrapper&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)


<!-- feel free to add additional comments -->
TODO:

- [x] Create a Class component wrapper for handling the drawer events
- [x] Create a Storybook to demonstrate the functional drawer

gilad215 commented:
After discussing with @sharvit (he explained in depth #264) , i'm creating 3 different storybooks for our Notification drawer:

1. Built with Components w/Children
2. Built with a "Wrapper" w/Props
3. Built with a Stateful "Wrapper" w/Props that makes the drawer functional

So far i created 1, 2

gilad215 commented:
Created a `StatefulDrawerWrapper`  and learned a lot from it ! 💯 
the Drawer is fully functional, and changes based on the notifications state.
(e.g. Reading all notifications in a certain panel will remove the "Mark all Read" Button, Reading all notifications from all panels will change the bell icon)

About CSS, I noticed i can only render the drawer correctly if its in a Nav bar, i added some inline css to the bell and the nav bar for now but any feedback about this will be appreciated.

gilad215 commented:
- Added missing Props comments from some components
- Added a Collapse animation for the Stateful wrapper panels
- Added a `showLoading` prop to the PanelWrapper in order to render a loading notification at the end of the list

AllenBW commented on src/components/Notification/Stories/BasicNotifDrawerStory.js:
```diff
@@ -0,0 +1,176 @@
+        <nav className="collapse navbar-collapse">
+          <ul className="nav navbar-nav navbar-right navbar-iconic">
+            <li className="drawer-pf-trigger open">
+              <a href="#0" className="nav-item-iconic" onClick={handleClick}>
```
I'm wondering...  would href="#" bring yah to the top of page given line 13?  Is `#0` necessary here?

gilad215 replied:
Hi Allen, i copied the Navbar html becuase there is no react implementation, `handleClick()` prevents the default function so clicking on the bell in this specific story won't do a thing.
anyway i can remove the `href`

AllenBW replied:
I guess the real question is... does the href of an a register as an event interface... I know mdn hates re-purposing anchors as buttons... https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Notes ...

gilad215 commented:
- Added Tests

gilad215 commented:
- Added Tests
- Seperated the Stateful Drawer from the Toggle, which also needs to be Stateful. This gives more freedom for the consumer so he can implement the opening/closing and `hasUnreadMessages` behaviour of the drawer.

ohadlevy commented:
@gilad215 is this still WIP? in case you are ready for review, please remove the WIP.

gilad215 commented on .gitignore:
```diff
@@ -18,5 +18,5 @@ coverage
 !.vscode/settings.json
 !.vscode/tasks.json
-!.vscode/launch.json
+.vscode/launch.json
```
Why should i add my launch.json?
a file with exclamation mark means gitignore ignores it if i'm not mistaken

priley86 replied:
if you are debugging inside vscode and would like to share your launch.json w/ others i think that's ok...i don't see the file included in your changeset though?

ohadlevy commented on src/components/Notification/NotifDrawer/NotifDrawerTitle.js:
```diff
@@ -0,0 +1,49 @@
+
+  return (
+    <div className={classes} {...props}>
+      <a
```
maybe use icon component? also is it possible to have the expansion optional ? (via a prop?)

gilad215 replied:
The expand toggle already has an icon in the `<a>` class, i added an icon for the close button.
as for the expansion, there is a prop for it in the `NotifDrawer`, I'll tag you.

gilad215 commented on src/components/Notification/NotifDrawer/NotifDrawer.js:
```diff
@@ -0,0 +1,34 @@
+const NotifDrawer = ({ hide, expanded, children, className, ...props }) => {
+  const classes = ClassNames(
+    'drawer-pf drawer-pf-notifications-non-clickable',
+    { 'drawer-pf-expanded': expanded },
```
@ohadlevy

ohadlevy replied:
wouldnt this just decide if its expanded or not? i was referring to not having the expand option, if you dont have a lot of content.

gilad215 replied:
You want the expand button to be optional? on the way..

gilad215 commented:
- Made the Drawer expandability optional, theres a knob in the stateful storybook

gilad215 commented:
Thanks @serenamarie125 !, will do.
Until then, Do you mind reviewing #212 ? You requested changes.

priley86 commented on package.json:
```diff
@@ -20,6 +20,7 @@
   "dependencies": {
     "breakjs": "^1.0.0",
     "classnames": "^2.2.5",
+    "lodash": "^4.17.5",
```
i don't see lodash referenced any place? did you mean to remove it?

priley86 replied:
would be great to stay lodash free in the distributed code if possible... we have been using it as `optionalDependency` for some of the Storybook story examples only so far...but trying to to stay es6 ;)
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore

priley86 commented on src/components/Notification/Notification.js:
```diff
@@ -0,0 +1,43 @@
+  if (loading) {
+    return (
+      <div className="drawer-pf-notification text-center">
+        <Spinner inline loading size="xs" /> Loading More
```
`Loading More` can be passed as a defaultProp (message) for i18n purposes...

priley86 replied:
(see PaginationRow for example if you want...)

priley86 commented on src/components/Notification/Notification.test.js:
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
```
react-test-renderer is slowly going away ;)... Enzyme gives us similar functionality and more fine grained tests... 

see:
https://github.com/patternfly/patternfly-react/pull/258

priley86 commented:
again - excellent work @gilad215 🏅 

The components are well separated, flexible, and you were able to use many of the existing PF React components, so excellent job there!

We definitely need some more review help @jeff-phillips-18  @Rohoover  @jgiardino @maryclarke. 

This is a pretty big one - maybe whoever is consuming this one first on the application side can also help review/integrate?

i'm pumped about this one coming to PF - React ! 👏 👏 👏 

also - i really liked the Foreman Community Newsletters you cite 🤣 😆

gilad215 commented:
- Switched to `enzyme` testing and checks are successful as of now 👍 @priley86 
- Added the ability to hide a certain notification through the drop down menu

jeff-phillips-18 commented on src/components/Notification/NotificationDrawer/NotificationDrawerTitle.js:
```diff
@@ -0,0 +1,61 @@
+    return (
+      <div className={classes} {...props}>
+        <a className="drawer-pf-toggle-expand" onClick={onExpandClick} />
+        <a className="drawer-pf-close" onClick={onCloseClick}>
```
rather than duplicating all but this, make this piece conditional:
```
{expandable && (
        <a className="drawer-pf-toggle-expand" onClick={onExpandClick} />
)}
```

jeff-phillips-18 commented on src/components/Notification/NotificationDrawer/NotificationDrawerToggle.js:
```diff
@@ -0,0 +1,40 @@
+      <Button onClick={onClick} bsStyle="link">
+        <Icon
+          name={iconName}
+          style={{ color: 'rgb(240, 240, 240)', marginTop: '20px' }}
```
this shouldn't be added. Patternfly has styling for the toggle while inside the mastheads. This will override them.

gilad215 replied:
I mentioned this before,  there is a CSS problem with the toggle. thats why i separated the `StatefulDrawerWrapper` from the toggle. i created this toggle for the demonstration only, ill try to find a better solution until the CSS experts arrive

jeff-phillips-18 replied:
I see. Adding it in the storybook code would be OK by me. If/when we have a masthead component it won't be necessary (I would hope).

jeff-phillips-18 replied:
@gilad215 I believe adding the `nav-item-iconic` class to the Button will get us the correct patternfly styling without the need for the inline styles.

jeff-phillips-18 commented on src/components/Notification/NotificationDrawer/NotificationDrawerToggle.js:
```diff
@@ -0,0 +1,40 @@
+          name={iconName}
+          style={{ color: 'rgb(240, 240, 240)', marginTop: '20px' }}
+          aria-describedby="tooltip"
+          size="lg"
```
the `lg` should not be added, the mastheads determine the font size.

gilad215 replied:
Removing the `size` makes the bell icon too small, any recommendations?
I can make this component a "Storybook Only" component, by naming it and using it only there

jeff-phillips-18 commented on src/components/Notification/NotificationInfo.js:
```diff
@@ -0,0 +1,28 @@
+    <div className={classes}>
+      <span className="date">{date}</span>
+      <span className="time">{time}</span>
+    </div>
```
This is specific to displaying the date and time. Applications may want to display other information here. Not sure if we should have a NotificationDateTimeInfo component for this (which seems common) case and have NotificationInfo be generic or handle children here and allow for date/times to be null.

gilad215 replied:
the spans need those classnames to render correctly, I can create `NotificationLeftText`,`NotificationRightText` components that will use those classnames.

jeff-phillips-18 commented on src/components/Notification/Wrappers/DrawerWrapper.js:
```diff
@@ -0,0 +1,96 @@
+import React from 'react';
```
Can we name this `NotificationDrawerWrapper` to be consistent? Same for the rest of the files under Wrappers.
NotificationDrawerPanelWrapper, NotificationDrawerStatefulWrapper etc...

Also, should we pull this all under NotificationDrawer? i.e. `NotificationDrawer.Wrapper`, `NotificationDrawer.StatefulWrapper`, ...

gilad215 replied:
Not sure the wrappers should be under, your call.

jeff-phillips-18 commented on src/components/Notification/Wrappers/DrawerWrapper.js:
```diff
@@ -0,0 +1,96 @@
+  /** on togglePanel Click Func */
+  togglePanel: PropTypes.func,
+  /** is Drawer Expandable */
+  isExpandable: PropTypes.bool
```
Comments for functions should list any received parameters.

jeff-phillips-18 commented on src/components/Notification/Wrappers/PanelWrapper.js:
```diff
@@ -0,0 +1,188 @@
+  onNotificationHide: PropTypes.func,
+  /** on MarkAllClear Click Func */
+
+  onMarkPanelAsClear: PropTypes.func,
```
comments for arguments on functions

gilad215 commented on src/components/Notification/NotificationDrawer/NotificationDrawerAccordion/NotificationDrawerPanelCollapse.js:
```diff
@@ -0,0 +1,36 @@
+}) => {
+  const classes = ClassNames(
+    'panel-collapse',
+    { 'collapse in': collapseIn },
```
I thought its necessary for the Collapse animation to behave correctly, but its working fine without it so i'm removing.

gilad215 commented:
Implemented most of @jeff-phillips-18 requests. Need more info on how to fix `NotificationInfo`.
Regarding the `NotificationDrawerToggle` - removing the style makes it almost invisible in the storybook, we need it just for the storybook but the fact that its a component contradicts the guidelines?

Another thought, what about creating `NotificationDrawerHeightMatching` @priley86? 🥇
Right now the drawer uses maximum height, and there is wasted space. surprisingly some people i asked thought its better this way, but maybe we need to change the drawer height dynamically like we did with the cards?

jeff-phillips-18 commented:
The design of the drawer has always been to take the full window height, though I don't see that specifically called out in the design at http://www.patternfly.org/pattern-library/communication/notification-drawer/#design.

@patternfly/patternfly-react-ux any thoughts?

gilad215 commented:
@jeff-phillips-18 @patternfly/patternfly-react-ux 
The Notification drawer in Foreman is changing its height dynamically ([pastebin of the CSS](https://pastebin.com/widbGqf8))

jeff-phillips-18 commented on src/components/Notification/Stories/basicNotificationDrawerStory.js:
```diff
@@ -0,0 +1,192 @@
+    const story = (
+      <nav className="navbar navbar-pf-vertical">
+        <nav className="collapse navbar-collapse">
+          <ul className="nav navbar-nav navbar-right navbar-iconic">
```
Any reason for not using NotificationDrawerToggle here?

jeff-phillips-18 commented on src/components/Notification/NotificationDrawer/index.js:
```diff
@@ -0,0 +1,27 @@
+NotificationDrawer.Title = NotificationDrawerTitle;
+NotificationDrawer.Dropdown = NotificationDrawerDropDown;
+
+export { NotificationDrawer, NotificationDrawerToggle };
```
Should we have NotificationDrawer.Toggle?

serenamarie125 commented on src/components/Notification/Stories/basicNotificationDrawerStory.js:
```diff
@@ -0,0 +1,192 @@
+                    Notification Tab 1
+                  </a>
+                </NotificationDrawer.PanelTitle>
+                <NotificationDrawer.PanelCounter text="3 New Events" />
```
@gilad215 the [Notification Drawer Design Pattern](http://www.patternfly.org/pattern-library/communication/notification-drawer/#design) was recently updated to status "# Unread Events" rather than "# New Events" in the accordion.  Can you please update this?

serenamarie125 commented on src/components/Notification/Notification.js:
```diff
@@ -0,0 +1,44 @@
+            ' Loading More'
+          ]
+        : children}
+    </div>
```
In your storybook, it seems like there is a missing horizontal line under the spinner:
<img width="325" alt="screen shot 2018-03-21 at 7 22 14 pm" src="https://user-images.githubusercontent.com/12504403/37744531-ab22c13a-2d3d-11e8-8af4-6018c095dd86.png">
below is the core example:
<img width="327" alt="screen shot 2018-03-21 at 7 22 04 pm" src="https://user-images.githubusercontent.com/12504403/37744529-a6965e06-2d3d-11e8-85c0-31bb168b3524.png">

serenamarie125 commented:
The vertical scroll is supposed to be specific to the content in the open accordion.  The scrolling works differently in this storybook
<img width="316" alt="screen shot 2018-03-21 at 7 38 11 pm" src="https://user-images.githubusercontent.com/12504403/37744845-6e268044-2d3f-11e8-9c93-7805e1cdd8aa.png">

gilad215 commented:
- Added `NotificationDrawerToggle` to imports and to the stories @jeff-phillips-18 
- Added scrolling and separated the read/clear buttons from the panel's body, this also fixed the missing horizontal line in the loading notification @serenamarie125 

Had to add this css for the scrolling to work, not sure why it didn't import it from core
```.drawer-pf {
  .panel-group {
    .panel-body {
      max-height: 364px; //  4 notifications height
      overflow-y: auto;
    }
  }
}

jeff-phillips-18 commented:
@gilad215 I'm not sure where in core you are expecting to get those settings on the panel-body from. The max-height and overflow settings are set by the javascript that is in place when using the `initCollapseHeights` function for the fixed height accordion. We should do similar here as the value should be computed rather than hard coded.

gilad215 commented:
@jeff-phillips-18 Panels now consume maximum height of the window as default. Added `maxPanelHeight` prop to allow a custom height, this will enable inner scrolling if necessary.

gilad215 commented:
@jgiardino I'm not sure what you mean by "custom fixed height is taller than the window", This change is suppose to make it easier for us to define the drawer. as @jeff-phillips-18 said, the default behaviour should be the panels consuming all the space they can, meaning the whole window, this is fine with full window drawers, but if we take a look at the Foreman drawer, they set the drawer max-height to 600px, and with the default behaviour the drawer looks like this (no read,clear buttons and no scrolling)

<img width="402" alt="screen shot 2018-03-30 at 0 54 15" src="https://user-images.githubusercontent.com/24938324/38115719-af343082-33b5-11e8-8147-f864dce4b8b2.png">

unless they change the drawer height to consume full window ( i don't think we should force them to ) we should give them an option to change the panel height ( their responsibility to handle layout issues ) for example 350px and it looks much better:

<img width="344" alt="screen shot 2018-03-30 at 1 08 11" src="https://user-images.githubusercontent.com/24938324/38115990-db683558-33b6-11e8-938c-ce48843ea00c.png">

Again, i'm no CSS expert, Foreman implemented their [own CSS](https://pastebin.com/k8um6LNP) to make the drawer change height dynamically, there could be a better solution to this issue and ill be glad to learn. Still, i thinks this add flexibility and not complexity

jeff-phillips-18 commented:
The setting of the maxHeight by the application becomes their responsibility. If set too large or too small there will be issues but these should not, and likely cannot, be handle by the patternfly react component. I'm OK with letting the application handle the setting of the maxHeight. We would need to verify that the component adjusts correctly if/when the property is updated.

gilad215 commented:
@jeff-phillips-18 @jgiardino  sounds great to me, to verify the component updates correctly i added a `maxHeight` [KNOB](http://localhost:6006/?knob-Expanded%20Drawer=false&knob-is%20Drawer%20Expandable=true&knob-Panel%20Height=364&selectedKind=NotificationDrawer&selectedStory=Stateful%20Drawer%20Wrapper&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs) to the stateful storybook.

jgiardino commented:
Thanks for clarifying and adding the KNOB, @gilad215! I was confused about what the Panel Height property was setting, and thought it was setting the full drawer height. 

When I look at the storybook example, I'm not able to reproduce the examples that you show. Are the examples you show what's currently implemented in Foreman? 

Just to make sure I'm on the same page as you guys... 

- In PF core, we dynamically set the max-height of the panel based on the window height. 
  - This results in a broken layout if the application sets the height to be something other than based on the window height.
  - I also noticed that the calculated value seems a little off at times.
- in this PF-react example, we do not dynamically set the max-height of the panel, but we do provide the ability to set the panel height.
  - Is the expectation that if applications are setting the drawer height, as in Foreman, then they would dynamically set the panel height based on the drawer height?

I included below some of the examples I'm able to produce in the storybook demo by setting the Panel Height KNOB and also using the inspector to change the total height on the drawer.

I'm curious what everyone thinks is the preferred solution in this case. I would think that the ideal solution is to have the component dynamically set the panel height based on the height of the drawer. But if that isn't an option here, then I'm wondering if we should even provide the panel height property at all. The best case scenario with this option is for applications to implement the behavior to dynamically set the panel height, but there's potential for this to be handled inconsistently across applications and result in weird double scrolling behavior as shown below. Whereas the examples below where the panel height isn't set seem like a better alternative.

Let me know what you think is the right solution. I ultimately just want to make sure we understand what our expectations are for applications if we are deviating from what core provides and documents.

Total Height is 400px
Panel Height is not set
![image](https://user-images.githubusercontent.com/21063328/38197271-6842db7c-3655-11e8-9675-4d070cac34ec.png)

Total Height is calculated from window height
Panel Height is not set
![image](https://user-images.githubusercontent.com/21063328/38197408-e6a43d12-3655-11e8-94c1-c728b24b45ac.png)

Total Height is 400 px
Panel Height is 225px
![image](https://user-images.githubusercontent.com/21063328/38197299-7c29eb80-3655-11e8-8708-2cc72cba7d40.png)

Total Height is calculated from window height
Panel Height is 225px
![image](https://user-images.githubusercontent.com/21063328/38197347-a7154d08-3655-11e8-8c32-6e8be77f6638.png)

gilad215 commented:
@jgiardino Thanks for your input.

Confirming: If a drawer is not full window height, a panel without a custom height can cause a broken layout

Interesting enough,In Foreman the drawer is changing its height based on the panel height, It has a **max-height for the drawer** and thats why the broken layout appears, but if the height of the panels is smaller than the max-height of the drawer (with the help of the `maxPanelHeight`) it looks nice without any wasted space. 
You are suggesting the panel will change its height based on the drawer height.
I'm not sure which way is better, but i would love to hear some CSS suggestions on this, I think dynamic height changing can improve the drawer even more!
Maybe we can learn something from their [CSS](https://pastebin.com/k8um6LNP)?

jgiardino commented:
> Confirming: If a drawer is not full window height, a panel without a custom height can cause a broken layout

Do you have a screen capture of this component, showing this case? I included examples above of what I can produce from the storybook. But the examples that appear to be broken are the cases where the panel height is set.

> You are suggesting the panel will change its height based on the drawer height.

This was an assumption that I was making. PF core currently dynamically sets the panel height, and so I assumed that the expectation was for applications to dynamically set the panel height using the panel height property you provided. Otherwise, if the panel height is static, then there will be cases where two scrollbars display, as shown in my 3rd example above.

There is currently a [related design issue](https://github.com/patternfly/patternfly-design/issues/573) that is open. Whatever the decision is for this would impact the css implementation. But until then, I'm not sure what the best option is. 

It may be that for the use case in Foreman, the ability to set a panel height to a fixed pixel size works well. But my concern is that this solution is very specific for that use case, and might have really bad results for other applications. And I totally get the argument that the consumers should take responsibility for properties that they set, but my instinct still is that we should provide something that will always resolve to a good layout ;-) (either that, or at least note that this property could result in unwanted effects on the layout and scrolling of the drawer)

jeff-phillips-18 commented:
I believe if the maxHeight is not set, we should do the same as PF core is doing in the `initCollapseHeights` function for the fixed height accordion.

gilad215 commented:
@jeff-phillips-18 @jgiardino 
- Removed `maxPanelHeight` prop, Panels now change their height dynamically 🥇  by
creating an event listener that observes any window resize, and changes the `accordionHeight` prop

Then the Panel body height is calculated:
`accordionHeight - (panelCount * closedPanelHeight)`

![paneldynamic](https://user-images.githubusercontent.com/24938324/38245616-860dbabc-3747-11e8-839f-91e045b794d0.gif)

This functionality requires a State, consuming the `NotificationDrawerWrapper` alone won't have this dynamic height changing.
I guess we can approach this in two ways, either the consumer copies my functions `componentDidMount`, `componentWillUnmount`, `updateAccordionHeight` to his Stateful wrapper, Or we can let him Override the Stateful wrapper default functions with props.

jgiardino commented on src/components/Notification/Wrappers/PanelWrapper.js:
```diff
@@ -34,6 +35,8 @@ const PanelWrapper = ({
     return '1 Unread Event';
   };
 
+  const getPanelHeight = () => accordionHeight - (panelCount * 55 + 41);
```
The numbers 55 and 41 are very specific and could be different in different rendering contexts. Is it possible to capture this value when the component is rendered?

@jeff-phillips-18 is that something we would want to do in this case (i.e. get actual height of drawer header and panel header to use here)?

jeff-phillips-18 replied:
Yeah, we can't assume the heights of the heading and footer. Each could be different based on contents.

jgiardino replied:
The other thing I'm noticing is that the panel header heights could vary depending on the length of the text string (e.g. if it wraps) and also the state of the drawer being normal width or wide width (e.g. a long string wraps at normal width but doesn't at wide width).

gilad215 replied:
Right,
I think i can keep going with `getElementsByClassName`, since it returns an array of the findings (each of them can have different height), i can get the exact height we need for the panel. ill update tonight

gilad215 commented:
@jeff-phillips-18 @jgiardino 
All the heights calculations are now happening in the state. It searches for all the panel headings calculates their height dynamically. We also need to subtract the Read/Clear section height, do you think we need to search the DOM for it? I think we should declare its height (44px) as a const like this:

```panelHeight =document.getElementsByClassName('panel-group')[0].clientHeight - READ_CLEAR_HEIGHT;```

gilad215 commented:
- Removed the JS Dynamic height in favour of CSS thanks to @jeff-phillips-18[ PR](https://github.com/patternfly/patternfly-react/pull/262/commits/13a0c99664bc2164664f0ebbcb499fba982ad9c7) 🥇

jgiardino commented on sass/patternfly-react/_notificationdrawer.scss:
```diff
@@ -0,0 +1,45 @@
+      display: flex;
+      flex-direction: column;
+      flex: 1 1 auto;
+
```
If you add `min-height: 0;`, this will fix a scrolling issue that is occurring in Firefox.

jgiardino commented on sass/patternfly-react/_notificationdrawer.scss:
```diff
@@ -0,0 +1,45 @@
+  .drawer-pf-action-link {
+    align-self: center;
+    margin: 0;
+  }
```
I would replace all of this:
```
 .drawer-pf-action {
    padding: 10px 0;
  }

  .drawer-pf-action-link {
    align-self: center;
    margin: 0;
  }
```
with this:
```
  .drawer-pf-action {
    flex: none;
  }
```
We don't need this item to be a flex item. So setting `flex: none;` let's us use the spacing as it's defined in core without having to recreate it a different way.

priley86 commented on src/index.js:
```diff
@@ -21,6 +21,7 @@ export * from './components/MenuItem';
 export * from './components/Modal';
 export * from './components/Nav';
 export * from './components/Navbar';
+export * from './components/Notification';
```
do you mean to `export * from './components/NotificationDrawer'` as well?

priley86 commented:
wow... 73 commits. you are a trooper @gilad215 🎖 

as a first round, this looks awesome... approving.
